src/libostree/ostree-bootloader-uboot.c

```diff
@@ -95,7 +95,45 @@ create_config_from_boot_loader_entries (OstreeBootloaderUboot     *self,
 
   val = ostree_bootconfig_parser_get (config, "options");
   if (val)
-    g_ptr_array_add (new_lines, g_strdup_printf ("bootargs=%s", val));
+    {
+      glnx_fd_close int uenv_fd = -1;
+      g_autofree char* kargs = g_strdup (val);
+      const char *uenv_path = NULL;
+      const char *ostree_arg = NULL;
+
+      g_ptr_array_add (new_lines, g_strdup_printf ("bootargs=%s", val));
+
+      /* Append system's uEnv.txt, if it exists in $deployment/usr/lib/ostree-boot/ */
+      ostree_arg = strtok (kargs, " ");
```

Comment:
I was going to say to use `strtok_r` here (since ostree is a threadsafe library, and other libraries may use `strtok`)...but then I think we really want to use `_ostree_kernel_args_from_string(val)` + `_ostree_kernel_args_get_last_value (kargs, "ostree")` or so.